### PR TITLE
Tap redis expiry checkers (no-data-received)

### LINKED ISSUE #478: Automatically restart decoupled monitor taps when redis key expires like in old monitor super-container
**Is your feature request related to a problem? Please describe.**
We are missing this (not critical though) feature we had in the initial monitor container
(after #472 is merged)

**Describe the solution you'd like**
Sth along the lines of the old code but adjusted to the decoupled arch

**Additional context**
Old code:
```
        def setup_redis_mon_listeners(self):
            def redis_event_handler(msg):
                if "pattern" in msg and "channel" in msg and "data" in msg:
                    monitor_to_restart = None
                    for pubsub_mon_channel in self.redis_pubsub_mon_channels:
                        if (
                            str(msg["channel"].decode()) == pubsub_mon_channel
                            and str(msg["data"].decode()) == "expired"
                        ):
                            match_monitor = re.match(
                                r"^.*\:(.*)_seen_bgp_update$",
                                str(msg["channel"].decode()),
                            )
                            if match_monitor:
                                monitor_to_restart = match_monitor.group(1)
                                break
                if monitor_to_restart and self.flag:
                    try:
                        # first, check if this monitor is currently running
                        proc_id_to_terminate = None
                        for proc_id in self.process_ids:
                            name, pid = proc_id
                            if name.startswith("[{}]".format(monitor_to_restart)):
                                log.info(
                                    "{} needs restarting".format(monitor_to_restart)
                                )
                                # stop it
                                proc_id_to_terminate = proc_id
                                try:
                                    pid.terminate()
                                except ProcessLookupError:
                                    log.exception("process terminate")
                                break
                        # start it
                        if proc_id_to_terminate:
                            self.process_ids.remove(proc_id_to_terminate)
                            init_mon_instance = getattr(
                                self, "init_{}_instance".format(monitor_to_restart)
                            )
                            init_mon_instance()
                    except Exception:
                        log.exception("Exception")

            try:
                self.redis_pubsub = self.redis.pubsub()
                self.redis_pubsub_mon_channels = [
                    "__keyspace@0__:ris_seen_bgp_update",
                    "__keyspace@0__:bgpstreamlive_seen_bgp_update",
                    "__keyspace@0__:exabgp_seen_bgp_update",
                    "__keyspace@0__:bgpstreamkafka_seen_bgp_update",
                ]
                for pubsub_mon_channel in self.redis_pubsub_mon_channels:
                    self.redis_pubsub.psubscribe(
                        **{pubsub_mon_channel: redis_event_handler}
                    )
                self.redis_listener_thread = self.redis_pubsub.run_in_thread(
                    sleep_time=1
                )
            except Exception:
                log.exception("Exception")
```

### PR DESCRIPTION
**Pull request by vkotronis**
### Description of PR

What component(s) does this PR affect?

- [ ] Back-End (Database, Detection/Configuration/etc. Microservices)
- [ ] Front-End (Flask, API, UI, etc)
- [X] Monitor (RIPE RIS, BGPStream RV/RIS/CAIDA, etc.)
- [ ] Docs
- [ ] Build System

Does the PR require changes on other components? If yes, please mark the components:

- [ ] Back-End (Database, Detection/Configuration/etc. Microservices)
- [ ] Front-End (Flask, API, UI, etc)
- [ ] Monitor (RIPE RIS, BGPStream RV/RIS/CAIDA, etc.)
- [ ] Docs
- [ ] Build System

### Related Issue

Resolves #478

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] None of the above

### Checklist:

- [X] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by vkotronis** [comment]
@slowr before reviewing the current PR check https://github.com/FORTH-ICS-INSPIRE/artemis/issues/558 since it is blocking (we need to read the redis conf correctly before we have the expiry subscription)

File: monitor-services/bgpstreamkafkatap/core/bgpstreamkafka.py [hunk 1]
```diff
@@ -364,6 +362,42 @@ def start_rest_app(self):
         IOLoop.current().start()
 
 
+class RedisChecker:
```
Comment (slowr) [inline monitor-services/bgpstreamkafkatap/core/bgpstreamkafka.py:365]: this would make sense to be in the `artemis-utils` package right?

Comment (vkotronis) [reply monitor-services/bgpstreamkafkatap/core/bgpstreamkafka.py:365]: You are right I will move it there

Comment (vkotronis) [reply monitor-services/bgpstreamkafkatap/core/bgpstreamkafka.py:365]: But how do I pass the stop_data_worker function which is custom? I should add it to the args probably and simply use it.

Comment (vkotronis) [reply monitor-services/bgpstreamkafkatap/core/bgpstreamkafka.py]: @slowr done it works OK, please re-check PR when available and let me know. The travis tests are not failed, they have reached pull rate limits (will re-issue them later)

**Comment by vkotronis** [comment]
@slowr any comments/objections on this PR? I reused code as discussed by importing from utils.

**Review by slowr — APPROVED** [review]

**Pull request merged**